[#752] Update Dropdown component

planktonic opened the pull request:
Fixes #752 

## Changes

- Uses the Button component in the dropdown stories, and adds a Dropdown component for use in these stories
- Updates appearance of dropdowns to match Zeplin
- button--menu now has no border-radius

## Looks like

<img width="1100" alt="Screenshot 2023-04-04 at 14 23 24" src="https://user-images.githubusercontent.com/2479422/229790431-b90962fa-d2f4-4d66-9075-32e4bd552fef.png">

## How to QA Visual changes

- `git checkout main` 
- `git pull` 
- `git checkout feature/752-update-dropdown-component` 
- `yarn`
- `yarn storybook`
- navigate to atoms/dropdown

Check:

- [x] The appearance matches the zeplin designs. 
- [x] The documentation for this component is correct, understandable, and up-to-date.

## How to QA code changes

- [x] The documentation for this component is correct, understandable, and up-to-date.
- [x] The component makes good use of CSS custom properties to simplify creating variants (or doesn’t have variants).
- [x] Everything that should be a variable, is.

## Preflight checks

PR author to check, and delete if not applicable:

- [x] Storybook documentation has been updated
- [x] Fixtures in [`test/scss/`](../test/scss/) have been updated
- [x] Your changes have been added to the `unreleased` section of [CHANGELOG.md](../CHANGELOG.md)

ed-eva reviewed (commented):
Looks good!
A couple points:
* There is a spacing of 4px at the top and bottom of the dropdown (between the buttons and the outline). There should be no spacing here. There is only spacing between items if there is a dividing line.
* Shadow should be at 30% opacity with an 8px blur. 
* Documentation
    * Under Alignment the examples looks a little odd, are they meant to be the mobile variants?

planktonic commented:
@ed-eva thanks! I’ve addressed all your feedback (with the same caveat about shadows as in https://github.com/bitcrowd/bitstyles/pull/751#issuecomment-1495538276)
